src/backend/distributed/commands/table.c

```diff
@@ -376,14 +376,37 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand)
 		leftRelationId = IndexGetRelation(leftRelationId, missingOk);
 	}
 
-	/*
-	 * Normally, we would do this check in ErrorIfUnsupportedForeignConstraintExists
-	 * in post process step. However, we skip doing error checks in post process if
-	 * this pre process returns NIL -and this method returns NIL if the left relation
-	 * is a postgres table. So, we need to error out for foreign keys from postgres
-	 * tables to citus local tables here.
-	 */
-	ErrorIfAlterTableDefinesFKeyFromPostgresToCitusLocalTable(alterTableStatement);
+	if (processUtilityContext == PROCESS_UTILITY_SUBCOMMAND)
```

Comment:
I was thinking something like the following where `processUtilityContext` is only relevant for `AlterTableDefinesFKeyBetweenPostgresAndNonDistTable`. Does the below work fine?
```C
	if (processUtilityContext != PROCESS_UTILITY_SUBCOMMAND && 
		AlterTableDefinesFKeyBetweenPostgresAndNonDistTable(alterTableStatement))
	{
```

Reply:
probably yes, let me try

Reply:
17bc6100bc8d66a7c834babb10bba25fcbfc3609